packages/ember-data/lib/system/record-array-manager.js

```diff
@@ -99,37 +102,54 @@ export default Ember.Object.extend({
 
     forEach(recordArrays, function(array) {
       filter = get(array, 'filterFunction');
-      this.updateRecordArray(array, filter, typeClass, record);
+      this.updateFilterRecordArray(array, filter, typeClass, record);
     }, this);
+
+    if (this.liveRecordArrays.has(typeClass)) {
+      var liveRecordArray = this.liveRecordArrays.get(typeClass);
+      this._addRecordToRecordArray(liveRecordArray, record);
+    }
   },
   /**
     Update an individual filter.
 
-    @method updateRecordArray
+    @method updateFilterRecordArray
     @param {DS.FilteredRecordArray} array
     @param {Function} filter
     @param {subclass of DS.Model} typeClass
     @param {Number|String} clientId
   */
-  updateRecordArray: function(array, filter, typeClass, record) {
-    var shouldBeInArray;
+  updateFilterRecordArray: function(array, filter, typeClass, record) {
+    var shouldBeInArray = filter(record.getRecord());
+    var recordArrays = this.recordArraysForRecord(record);
 
-    if (!filter) {
-      shouldBeInArray = true;
+    if (shouldBeInArray) {
+      this._addRecordToRecordArray(array, record);
     } else {
-      shouldBeInArray = filter(record.getRecord());
+      recordArrays.delete(array);
+      array.removeRecord(record);
     }
+  },
 
+  _addRecordToRecordArray: function(array, record) {
     var recordArrays = this.recordArraysForRecord(record);
+    if (!recordArrays.has(array)) {
+      array.addRecord(record);
+      recordArrays.add(array);
+    }
+  },
 
-    if (shouldBeInArray) {
-      if (!recordArrays.has(array)) {
-        array.addRecord(record);
-        recordArrays.add(array);
+  updateLiveRecordArray: function(array, modelName) {
```

Comment:
This doesn't really update anymore, just initially populates it, so a different name/location might be better